frontend/catalyst/pennylane_extensions.py

```diff
@@ -2250,3 +2252,211 @@ def _callee():
 
     else:
         raise ValueError(f"Expected a callable or a qml.Operator, not {f}")  # pragma: no cover
+
+
+AxisSizeType = Union[int, jax.core.Tracer, jax.core.Var]
+
+
+def vmap(
+    fn: Callable,
+    in_axes: Union[int, Sequence[Any]] = 0,
+    out_axes: Union[int, Sequence[Any]] = 0,
+    axis_size: Optional[int] = None,
+) -> Callable:
+    """A :func:`~.qjit` compatible vectorizing map for PennyLane/Catalyst.
+
+    Args:
+        f (Callable): A Python function containing PennyLane quantum operations.
+        in_axes (Union[int, Sequence[Any]]): It specifies the value(s) over which input
+            array axes to map.
```

Comment:
yep, I didn't want to copy their docs. happy to add links to jax.vmap docs after hearing @josh146's opinion on this  :+1:

Reply:
I think the resolution from an offline chat with Josh was, no, we don't want to copy their docs, BUT our documentation needs to be sufficient for users to understand and use this function, without necessarily referring to JAX's documentation. So the descriptions here need to be a bit more detailed.

In order to avoid having these abstract descriptions be too long, it's also a good idea to use examples instead that show the different usage modes.